app/helpers/api/v1/simulator_helper.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Api::V1::SimulatorHelper
+  def return_image_file(data_url)
+    str = data_url[23..]
+    if str.to_s.empty?
+      path = Rails.public_path.join("images/default.png")
+      image_file = File.open(path, "rb")
+    else
+      jpeg = Base64.decode64(str)
+      image_file = File.new("tmp/preview_#{Time.zone.now.to_f.to_s.sub('.', '')}.jpeg", "wb")
```

Comment:
```suggestion
      image_file = Tempfile.new(["preview_", ".jpeg"], binmode: true)
```

This is suggested & improved code by AI, could you check it this works @Arnabdaz